homeassistant/bootstrap.py

```diff
@@ -552,6 +554,14 @@ def enable_logging(hass: core.HomeAssistant, verbose: bool=False,
 
         async_handler = AsyncHandler(hass.loop, err_handler)
 
+        @core.callback
+        def _async_close_handler(event):
+            """Close async handler on stop."""
+            async_handler.close()
+
+        hass.bus.async_listen_once(
+            EVENT_HOMEASSISTANT_STOP, _async_close_handler)
```

Comment:
I bet it only calls close when it has emitted something to the handler. 

I don't like this here because we won't see errors during shutdown.

What if we only start the thread on the first call to emit.

That way we know for sure that logging Will call close?